lib/compute/index.js

```diff
@@ -0,0 +1,1053 @@
+ *     `{name} {comparison} {filterString}`.
+ *     - **`name`**: the name of the field to compare
+ *     - **`comparison`**: the comparison operator, `eq` (equal) or `ne`
+ *       (not equal)
+ *     - **`filterString`**: the string to filter to. For string fields, this
+ *       can be a regular expression.
+ * @param {number} options.maxResults - Maximum number of addresses to return.
+ * @param {string} options.pageToken - A previously-returned page token
+ *     representing part of the larger set of results to view.
+ * @param {function} callback - The callback function.
+ *
+ * @example
+ * compute.getAddresses(function(err, addresses) {
+ *   // addresses is an array of `Address` objects.
+ * });
+ *
+ * //-
+ * // To control how many API requests are made and page through the results
+ * // manually, set `autoPaginate` to `false`.
+ * //-
+ * function callback(err, addresses, nextQuery, apiResponse) {
+ *   if (nextQuery) {
+ *     // More results exist.
+ *     compute.getAddresses(nextQuery, callback);
+ *   }
+ * }
+ *
+ * compute.getAddresses({
+ *   autoPaginate: false
+ * }, callback);
+ *
+ * //-
+ * // Get the addresses from your project as a readable object stream.
+ * //-
+ * compute.getAddresses()
+ *   .on('error', console.error)
+ *   .on('data', function(address) {
+ *     // `address` is an `Address` object.
+ *   })
+ *   .on('end', function() {
+ *     // All addresses retrieved.
+ *   });
+ *
+ * //-
+ * // If you anticipate many results, you can end a stream early to prevent
+ * // unnecessary processing and API requests.
+ * //-
+ * compute.getAddresses()
+ *   .on('data', function(address) {
+ *     this.end();
+ *   });
+ */
+Compute.prototype.getAddresses = function(options, callback) {
+  var self = this;
+
+  if (is.fn(options)) {
+    callback = options;
+    options = {};
+  }
+
+  options = options || {};
+
+  var path = '/aggregated/addresses';
+
+  this.makeReq_('GET', path, options, null, function(err, resp) {
+    if (err) {
+      callback(err, null, null, resp);
+      return;
+    }
+
+    var nextQuery = null;
+
+    if (resp.nextPageToken) {
+      nextQuery = extend({}, options, {
+        pageToken: resp.nextPageToken
+      });
+    }
+
+    var regions = resp.items || {};
+
+    var addresses = Object.keys(regions).reduce(function(acc, regionName) {
+      regionName = regionName.replace('regions/', '');
+
+      var region = self.region(regionName);
+      var regionAddresses = regions[regionName].addresses || [];
+
+      regionAddresses.forEach(function(address) {
+        var addressInstance = region.address(address.name);
+        addressInstance.metadata = address;
+        acc.push(addressInstance);
+      });
+
+      return acc;
+    }, []);
+
+    callback(null, addresses, nextQuery, resp);
+  });
+};
+
+/**
+ * Get a list of disks. For a detailed description of method's options see
+ * [API reference](https://goo.gl/M9Qjb3).
```

Comment:
For these ones you really will have to watch out for https://github.com/GoogleCloudPlatform/gcloud-node/pull/790